numpy/core/code_generators/generate_umath.py

```diff
@@ -431,42 +429,42 @@ def english_upper(s):
     Ufunc(2, 1, None,
           docstrings.get('numpy.core.umath.greater'),
           'PyUFunc_SimpleBinaryComparisonTypeResolver',
-          TD(all, out='?', simd=[('avx2', ints)]),
```

Comment:
Can you leave behind the `OO->?` loop, but put it at the end after OO->O? I think that would allow `np.equal(a, b, dtype=bool)` to be legal.

Reply:
This seems to be what @seberg was suggesting in the original issue